cpp/src/streaming/core/leaf_node.cpp

```diff
@@ -18,7 +18,9 @@ Node push_to_channel(
 
     for (auto& msg : messages) {
         RAPIDSMPF_EXPECTS(!msg.empty(), "message cannot be empty", std::invalid_argument);
-        co_await ch_out->send(std::move(msg));
+        if (!co_await ch_out->send(std::move(msg))) {
+            break;
+        }
     }
```

Comment:
if the channel is closed, this just breaks early rather than `send`ing remaining messages into a closed channel which won't do anything?